src/main/java/seedu/address/model/appointment/UniqueAppointmentList.java

```diff
@@ -21,4 +28,60 @@
     private final ObservableList<Appointment> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList.sorted(new AppointmentComparator()));
 
+    /**
+     * Returns true if the list has an overlapping appointment
+     */
+    public boolean overlaps(Appointment toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isOverlapping);
+    }
+
+    /**
+     * Adds an appointment to the list.
+     * The appointment must not overlap with existing appointments in the list.
+     */
+    public void add(Appointment toAdd) {
+        requireNonNull(toAdd);
+        if (overlaps(toAdd)) {
+            throw new OverlappingAppointmentException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Appointment> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    public void setAppointments(List<Appointment> appointments) {
+        requireAllNonNull(appointments);
+        if (!appointmentsAreNotOverlapping(appointments)) {
+            throw new OverlappingAppointmentException();
+        }
+
+        internalList.setAll(appointments);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof UniqueAppointmentList // instanceof handles nulls
+                && internalList.equals(((UniqueAppointmentList) other).internalList));
+    }
+
+    /**
+     * Returns true if {@code appointments} contains only non-overlapping appointments.
+     */
+    private boolean appointmentsAreNotOverlapping(List<Appointment> appointments) {
```

Comment:
would the method name be better if it sounds like a question?
like isAppointmentOverlapping

Reply:
Since I want it returns true when there is no overlapping. I think this name is somehow acceptable.